### ISSUE TITLE
Matching view for viewmodel not found

### ISSUE THREAD

**Issue by Revooo96** [Caliburn-Micro/Caliburn.Micro#387]
Hi,
I have a mainview and viewmodel which shows the currentviewmodel in the middle, i'm switching the currentviewmodel between different viewmodels 
the code for the switching should normally just work like this: currentviewmodel = new ...viewmodel(); and then it should find the matching view to the viewmodel, but when i debug the program it just shows the path where the current viewmodel is located --> it
 doesnt find the view for the viewmodel

![2016-11-07 12_27_02-](https://cloud.githubusercontent.com/assets/20872594/20056314/ccbab780-a4e5-11e6-85f1-187d53805f08.png)

i used the correct file system of caliburn
all views named with "view" in the end and are located in the folder "Views"
all viewmodelss named with "viewmodel" in the end and are located in the folder "Viewmodels"

anyone has an idea why it doesnt find the correct view? or did i miss something?

best regards

**Comment by nigel-sampson** [comment]
Typically when a view can't be found it's due to a refactor or move of the file name (location) but the underlying namespace wasn't changed.

I would check the namespace of both the view and view model is what you'd expect.

**Comment by Revooo96** [comment]
i already checked that aswell, it has nothing to do with namespace or something

**Comment by nigel-sampson** [comment]
Can you post the full namespaces of both the view and view model? or a sample illustrating the problem?

**Comment by Revooo96** [comment]
![2016-11-08 09_43_24-bueroautomatisation - microsoft visual studio](https://cloud.githubusercontent.com/assets/20872594/20092361/0fbc938e-a598-11e6-8a7f-24f9da2dc4e7.png)
![2016-11-08 09_44_14-bueroautomatisation - microsoft visual studio](https://cloud.githubusercontent.com/assets/20872594/20092363/0fc14384-a598-11e6-8c96-0bcd21a86662.png)
![2016-11-08 09_44_27-bueroautomatisation - microsoft visual studio](https://cloud.githubusercontent.com/assets/20872594/20092360/0fbc50fe-a598-11e6-9fda-efab8bc58d54.png)
![2016-11-08 09_44_52-bueroautomatisation - microsoft visual studio](https://cloud.githubusercontent.com/assets/20872594/20092362/0fc0b2a2-a598-11e6-8fb4-4ae1957d9822.png)

**Comment by nigel-sampson** [comment]
It's probably because the view model is private.

**Comment by nigel-sampson** [comment]
Spoke too soon, a private view model shouldn't affect view location (but possibly view model location).

Assume all other view / view model pairs are being located correctly? I can't see anything wrong with the screenshots. if you can provide a repro of the potential bug I'll dig into it further.

**Comment by nigel-sampson** [comment]
No further information provided.

**Comment by YJ15** [comment]
First of all, I apologize for not being good at English.

I had the same experience as you.
There are cases where the View cannot be found even though the rules of Caliburn.Micro are followed.
For example, let's say you create View and ViewModel as shown below.
-View name: MyView.xaml
-ViewModel name: MyViewModel.xaml

If you add a View or ViewModel to the project, you don't make the name MyView, MyViewModel, but make it different, and then change the name to MyView or MyViewModel.
Doing so can lead to an error where the View is not found.

Actually, when I ran into this problem, I recreated the View and ViewModel. When I entered the name correctly and created it, and I found the View normally.

In my opnion, it might be a bug in Caliburn.Micro or Visual Studio.
